### PR TITLE
[chore] dev mode refactor

### PR DESCRIPTION
**Pull request by Rich-Harris**
WIP. follow-up to #3129. Using Vite's watcher means we can push more logic into the plugin. Also, `Watcher` extends `EventEmitter` but we're not actually emitting any events, so we can make all that stuff way simpler

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/core/dev/index.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
-import { EventEmitter } from 'events';
 import fs from 'fs';
 import path from 'path';
 import { URL } from 'url';
 
 import { svelte } from '@sveltejs/vite-plugin-svelte';
 import amp_validator from 'amphtml-validator';
-import CheapWatch from 'cheap-watch';
```
Comment (benmccann) [inline packages/kit/src/core/dev/index.js:8]: this can be removed from the `package.json`

File: packages/kit/src/core/dev/index.js [hunk 2]
```diff
@@ -475,43 +322,113 @@ async function create_plugin(config, dir, https, get_manifest) {
...
-		configureServer(vite) {
-			return () => {
-				remove_html_middlewares(vite.middlewares);
-				vite.middlewares.use(create_kit_middleware(vite));
-			};
-		}
+		address_info,
+		server_config: vite_config.server,
+		allow,
+		cwd,
```
Comment (benmccann) [inline packages/kit/src/core/dev/index.js:405]: I'm not sure that `cwd` or `close` are used?

Comment (Rich-Harris) [reply packages/kit/src/core/dev/index.js:405]: `close` is (by the tests). i'm still working on this PR, it's probably not worth reviewing until it's marked ready — more changes to come

**Comment by Rich-Harris** [comment]
This will fix #2793

**Comment by Rich-Harris** [comment]
Bit weirded out by these deployment failures. Investigating

**Comment by Rich-Harris** [comment]
deploys all passing — issue turned out to be netlify being a bit overzealous with the build cache

**Review by benmccann — APPROVED** [review]

**Pull request merged**